source/adapters/level_zero/v2/command_list_cache.hpp

```diff
@@ -0,0 +1,63 @@
+//===--------- command_list_cache.hpp - Level Zero Adapter ---------------===//
+//
+// Copyright (C) 2024 Intel Corporation
+//
+// Part of the Unified-Runtime Project, under the Apache License v2.0 with LLVM
+// Exceptions. See LICENSE.TXT
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+#pragma once
+
+#include <stack>
+
+#include <ur/ur.hpp>
+#include <ur_api.h>
+#include <ze_api.h>
+
+#include "common.hpp"
+
+namespace v2 {
+struct immediate_command_list_descriptor_t {
+  ze_device_handle_t Device;
+  ZeStruct<ze_command_queue_desc_t> QueueDesc;
```

Comment:
all of QueueDesc? Won't that be a little annoying to use?

Reply:
I think we need all the fields from ze_command_queue_desc_t anyway (expect for index), no?

Reply:
This is false equivalence. This struct contains stuff we don't need (like pNext) that this will likely never use. I think we should specify each variable we need as part of the struct (or, which would be my preference, have a function like `getImmediateCommandList(params for immedaite stuff, all spelled out)`.

Reply:
Ok, I've changed it as you requested but that argument can be made for most places where we use/store L0 desc structures. Do we want to only create desc structure as late as possible (just before calling the L0 API call) and just have custom structures before that?

Reply:
> Do we want to only create desc structure as late as possible (just before calling the L0 API call) and just have custom structures before that?

If the intent is for the desc structure to be passed as-is to the L0 call, and if it more convenient, then I think its fine for it to be passed as an argument. However, if the structure is then somehow interpreted/manipulated by the adapter function, we should prefer to have our own structures/arguments, where we can define all the behavior for all the cases.
My concern here is that, when using L0 structs for non-L0 API calls, we potentially leave the API underdefined. For example, let's imagine a scenario where L0 has some sort of enum that lists supported usage models, but the L0 Adapter only supports a select number of those modes/behaviors. In this case, accepting the L0 type would make it seem as-if the API supports everything that the L0 specifies, which is not the case. In such scenarios, my preference would be for adapter to define its own types, that are aligned with that it supports.